### PR TITLE
30 failed rules info

### PR DESCRIPTION
**Pull request by seaparks**
Made these changes to hopefully fulfill: https://github.com/VividCortex/docs/issues/30

@potocnyj @mtrias - would you like to take a look? Pretty basic at the moment so if you want more information included, I can do that. Right now it's a change to the How to Use "Queries" page (just a reference to that we check for, and display the results of, rules) and a link to a new General Reference "Query Analysis Rules" page which just lists the things we check for (from core/rules).

### CODE REVIEW

**Comment by seaparks** [comment]
Oh hm. Let me see if I can figure out what to do about that ^

**Comment by seaparks** [comment]
Okay I'll need coaching on how to resolve that conflict. I'll get with @potocnyj

**Comment by mtrias** [comment]
Would it be possible to add a header for each of the rules, so we can link from the website?

**Comment by mtrias** [comment]
Otherwise looks good to me

**Comment by mtrias** [comment]
In order to resolve the conflicts, try using the web editor. i.e. click this link:
image

**Comment by seaparks** [comment]
@mtrias - we can, happy to do that. One thing though is that at the moment, the anchored links won't contain any more information than what's already going to be displayed in the UI, right? Would it be frustrating to click a link that takes you to a page with no new info?

I can try to write up a fuller description of those rules, but that might be an Ewen or other DB expert thing.

**Comment by mtrias** [comment]
Right, that makes sense. Only do that if / when it's worth the navigation :)

File: content/general-reference/query-analysis-rules.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Query Analysis Rules"
+maintitle: "Query Analysis Rules"
+docs: "reference"
+weight: "4"
+---
+
+Each query sample which VividCortex captures is analyzed to see whether it fails one (or more) of a number of rules,
```
Comment (potocnyj) [inline content/general-reference/query-analysis-rules.md]: I'd add the list of supported databases for query analysis rules at the top (currently just MySQL)

File: content/general-reference/query-analysis-rules.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+* Constant expressions in GROUP BY or ORDER BY clauses are at best useless operations that do not change query results.
+
+* GROUP BY or ORDER BY clauses with columns from different tables will force the use of a temporary table and filesort, which can be a huge performance problem and can consume large amounts of memory and temporary space on disk.
+
+* GROUP BY or ORDER BY clauses that sort the results in different directions prevent indexes from being used. All expressions in the ORDER BY clause must be ordered either ASC or DESC so that MySQL can use an index.
+
+* Selecting all columns with the * wildcard will cause the query’s meaning and behavior to change if the table’s schema changes, and might cause the query to retrieve too much data.
+
+* Evidence of SQL injection.
```
Comment (potocnyj) [inline content/general-reference/query-analysis-rules.md]: One thing worth mentioning is that this rule also generates a critical event when it is triggered.

**Pull request merged**